### PR TITLE
Azure OpenAI: update Completions finish reason representation for forward compat

### PR DESCRIPTION
**Pull request by trrwilson**
`CompletionsFinishReason` is currently represented as a nullable on `Choice` and `ChatChoice` to facilitate its reuse for deserialization of streaming `delta` messages. This is an implementation leakage, as the finish reason on final, non-streaming `Choice` and `ChatChoice` objects is never null.

This (breaking, in preview) change updates the representation of `CompletionsFinishReason` to align with intended API and with both near- and medium-term future updates to the typespec and functional set for AOAI.

### CODE REVIEW

**Review by glecaros — APPROVED** [review]

File: sdk/openai/Azure.AI.OpenAI/src/Custom/StreamingChoice.cs [hunk 1]
```diff
@@ -34,7 +34,7 @@ public class StreamingChoice
         /// provides "length." If no value is present, this StreamingChoice is still in progress.
         /// </remarks>
         public CompletionsFinishReason FinishReason
-            => GetLocked(() => _baseChoices.Last().FinishReason.Value);
+            => GetLocked(() => (CompletionsFinishReason)_baseChoices.Last().FinishReason);
```
[the remark this answers is not in the document]
Comment (trrwilson) [reply sdk/openai/Azure.AI.OpenAI/src/Custom/StreamingChoice.cs:37]: definitely not! thank you; I had my head so far up the "getting nullables to work" that I missed the simplification removing the type split entirely. fixed in the next commit.

File: sdk/openai/Azure.AI.OpenAI/src/Generated/AzureOpenAIModelFactory.cs [hunk 2]
```diff
@@ -52,9 +53,19 @@ public static EmbeddingsUsage EmbeddingsUsage(int promptTokens = default, int to
...
-        public static Choice Choice(string text = null, int index = default, CompletionsLogProbabilityModel logProbabilityModel = null, CompletionsFinishReason? finishReason = null)
+        public static Choice Choice(string text = null, int index = default, CompletionsLogProbabilityModel logProbabilityModel = null, CompletionsFinishReason finishReason = default)
         {
+            if (text == null)
+            {
+                throw new ArgumentNullException(nameof(text));
+            }
+            if (logProbabilityModel == null)
+            {
+                throw new ArgumentNullException(nameof(logProbabilityModel));
```
[the remark this answers is not in the document]
Comment (trrwilson) [reply sdk/openai/Azure.AI.OpenAI/src/Generated/AzureOpenAIModelFactory.cs:66]: Agreed, and it seems like a bug given this is the .tsp:
```
logprobs: CompletionsLogProbabilityModel | null;
```

[the remark this answers is not in the document]
Comment (trrwilson) [reply sdk/openai/Azure.AI.OpenAI/src/Generated/AzureOpenAIModelFactory.cs:66]: I'll move this to /custom to omit the extra null check.

File: sdk/openai/Azure.AI.OpenAI/src/Custom/StreamingChoice.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (trrwilson) [reply sdk/openai/Azure.AI.OpenAI/src/Custom/StreamingChoice.cs:37]: it absolutely should! updating in next iteration shortly.

**Review by joseharriaga — APPROVED** [review]

**Pull request merged**